libraries/ts-command-line/src/parameters/CommandLineDefinition.ts

```diff
@@ -62,6 +62,15 @@ export interface IBaseCommandLineDefinition {
    *   ordinary String Parameter:  Any value is accepted, including an empty string.
    */
   environmentVariable?: string;
+
+  /**
+   * This option can be used in cases where a command-line parameter may have
+   * synonyms that the application developer doesn't want to be visible in documentation.
+   * These can be useful when a command-line parameter is renamed, but the developer
+   * doesn't want to break backwards compatibility with systems still invoking the old
+   * longName for the parameter.
+   */
+  undocumentedSynonyms?: string[];
```

Comment:
Suggest to lead with the motivation for this feature, rather than the technical description:
```
  /**
   * Specifies additional names for this parameter that are accepted but not displayed
   * in the command line help.
   *
   * @remarks
   * This option can be used in cases where a command-line parameter may have been renamed,
   * but the developer doesn't want to break backwards compatibility with systems that may
   * still be using the old name. Only the `parameterLongName` syntax is currently allowed.
   */
```
I thought about how we might support this for `parameterShortName` in the future BTW. Seems like it would be okay for both long names and short names to eventually go in the same list.